nextmv/cloud/application.py

```diff
@@ -717,13 +719,13 @@ def _run_result(
             endpoint=f"{self.endpoint}/runs/{run_id}",
             query_params=query_params,
         )
+        result = RunResult.from_dict(response.json())
         if not large_output:
-            return RunResult.from_dict(response.json())
+            return result
 
-        download_url = DownloadURL.from_dict(response.json())
-        response2 = self.client.request(
-            method="GET",
-            endpoint=download_url.url,
-        )
+        download_url = DownloadURL.from_dict(response.json()["output"])
```

Comment:
What if we do `client2=copy.deepcopy(client)` and then `client2.headers = {}`? That way we can use `client2.request(...)`, the way it was?

Reply:
I thought about that, but it seemed a bit overcomplicated (for what it needs to do) and more brittle to me. After all, the pre-signed download url is a classic GET request.
But yeah, no strong opinion here. I assumed that your style may vary, so, feel free to change it. Just wanted to drop my motivation for choosing this style. :relaxed:

Reply:
Please see my other comment.

Reply:
On it. :100:

Reply:
Confirmed working. :+1: